### ISSUE TITLE
Python 3.12 support

### ISSUE THREAD

**Issue by JaGeo** [libAtoms/QUIP#645]
I am wondering if there is a chance for a Python 3.12 version.

**Comment by jameskermode** [comment]
Yup, we'd be happy to support this. Would you be willing to make a PR changing the wheel building script?
https://github.com/libAtoms/QUIP/blob/public/.github/workflows/build-wheels.yml

Once this works I'll be happy to tag a new release to PyPI.

**Comment by jameskermode** [comment]
Most likely just needs a version bump of cibuildwheel [here](https://github.com/libAtoms/QUIP/blob/public/.github/workflows/build-wheels.yml#L27)

**Comment by ATM-Jahid** [comment]
@jameskermode Please check PR #649.

**Comment by JaGeo** [comment]
@jameskermode sorry that I did not react! I overlooked the mentions for some reason! 

thanks @ATM-Jahid for working on this!

**Comment by jameskermode** [comment]
#649 is merged, but unfortunately the wheel builder doesn't run on PRs and I can now see it's failing, I think due to old manylinux containers not supporting Python 3.12 so we'll need to do a bit more work on this.

Logs: https://github.com/libAtoms/QUIP/actions/runs/9767786532/job/26963837065#step:5:283

**Comment by janosh** [comment]
just to note that https://github.com/materialsproject/atomate2/pull/992 is currently failing CI due to a missing `quippy-ase` wheel for 3.12:

>   ╰─▶ Because quippy-ase==0.9.14 has no wheels with a matching Python ABI tag
      and atomate2[strict-forcefields]==0.0.1 depends on quippy-ase==0.9.14...

i'll go ahead and restrict it for now 

```
"quippy-ase==0.9.14; python_version < '3.12'",
```

**Comment by jameskermode** [comment]
Noted, thanks. I don't think I will have the time to continue to update quippy for new Python releases, so restricting versions is the best thing to do for now. If anyone else would be willing to pick up from where @ATM-Jahid left off I'm sure it would be much appreciated.

**Comment by JaGeo** [comment]
Thanks @jameskermode. I will check if we can pick this up in the future. At least Python 3.12 support would be great.

**Comment by bernstei** [comment]
@jameskermode are you doing the updates manually, or can we have the github CI create them automatically for every python version that's available?

**Comment by jameskermode** [comment]
List is manual. But that's not the main problem, it's the version compatibilities discussed above which need real work to fix.

**Comment by ATM-Jahid** [comment]
Anyone trying to help with python 3.12 support should read [this comment](https://github.com/libAtoms/QUIP/pull/652#issuecomment-2207292226).

**Comment by JaGeo** [comment]
Just to follow up here: I am not sure if I completely understand the issue. I thought a migration to a new numpy version was needed for f90wrap but this seem to be done already. I am not really sure we can help.

**Comment by jameskermode** [comment]
No, f90wrap is already migrated, the problem is that the latest version of f90wrap is no longer compatible with quippy due to an inadvertent regression.

**Comment by JaGeo** [comment]
@jameskermode I have been wondering about this issue again. I have seen you have now fixed the workflows for Python 3.12: https://github.com/libAtoms/QUIP/actions/runs/20077050443/job/57594124503

Does this mean Python 3.12 support is now possible? Will there be a new version soon? 

Thanks in advance!

**Comment by jameskermode** [comment]
Yes - see #695 and the associated Mesonify branch for a preview. If you are willing to test that would be very useful in the run-up to releasing the new version on PyPI.

**Comment by JaGeo** [comment]
That's great. From the PR description it doesn't look like the interface has changed. Thus, I think we can manage to test it fairly quickly. Will try to give the installation at least a go this week

**Comment by jameskermode** [comment]
Resolved by merging #695. PyPI release coming up shortly.

**Comment by JaGeo** [comment]
@jameskermode thank you! Sorry for not following up. I did not manage to compile quippy from source but I am happy to test the new version in detail.
